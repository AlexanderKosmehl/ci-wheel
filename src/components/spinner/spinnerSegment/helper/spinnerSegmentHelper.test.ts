import { describe, expect, it } from '@jest/globals';
import {
  calculateSufficientHightLocation,
  convertRemToPixels,
  createSegmentClipPathString,
  createSegmentRotationString,
} from './spinnerSegmentHelper';

describe('spinnerSegmentHelper', () => {
  describe('createSegmentRotationString', () => {
    it('creates correct rotation strings', () => {
      expect(createSegmentRotationString(1, 1)).toMatchSnapshot();

      expect(createSegmentRotationString(1, 2)).toMatchSnapshot();
      expect(createSegmentRotationString(2, 2)).toMatchSnapshot();

      expect(createSegmentRotationString(1, 3)).toMatchSnapshot();
      expect(createSegmentRotationString(2, 3)).toMatchSnapshot();
      expect(createSegmentRotationString(3, 3)).toMatchSnapshot();

      expect(createSegmentRotationString(1, 4)).toMatchSnapshot();
      expect(createSegmentRotationString(2, 4)).toMatchSnapshot();
      expect(createSegmentRotationString(3, 4)).toMatchSnapshot();
      expect(createSegmentRotationString(4, 4)).toMatchSnapshot();
    });
  });

  describe('createSegmentClipPathString', () => {
    it('creates correct clippath strings', () => {
      expect(createSegmentClipPathString(1));
      expect(createSegmentClipPathString(2));
      expect(createSegmentClipPathString(3));
      expect(createSegmentClipPathString(4));
      expect(createSegmentClipPathString(5));
    });
  });

  describe('convertRemToPixels', () => {
    it('calculates pixels correctly', () => {
      expect(convertRemToPixels(0.25)).toBe(4);
      expect(convertRemToPixels(0.5)).toBe(8);
      expect(convertRemToPixels(0.75)).toBe(12);
      expect(convertRemToPixels(1.0)).toBe(16);
      expect(convertRemToPixels(1.25)).toBe(20);
    });
  });

  describe('calculateSufficientHightLocation', () => {
    it('calculates location correctly', () => {
      expect(calculateSufficientHightLocation(1)).toMatchSnapshot();
      expect(calculateSufficientHightLocation(2)).toMatchSnapshot();
      expect(calculateSufficientHightLocation(3)).toMatchSnapshot();
      expect(calculateSufficientHightLocation(5)).toMatchSnapshot();
      expect(calculateSufficientHightLocation(8)).toMatchSnapshot();
      expect(calculateSufficientHightLocation(13)).toMatchSnapshot();
      expect(calculateSufficientHightLocation(21)).toMatchSnapshot();
    });
  });
});
