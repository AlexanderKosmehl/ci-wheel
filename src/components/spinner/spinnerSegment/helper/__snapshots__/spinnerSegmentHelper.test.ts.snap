// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spinnerSegmentHelper calculateSufficientHightLocation calculates location correctly 1`] = `48`;

exports[`spinnerSegmentHelper calculateSufficientHightLocation calculates location correctly 2`] = `48`;

exports[`spinnerSegmentHelper calculateSufficientHightLocation calculates location correctly 3`] = `48`;

exports[`spinnerSegmentHelper calculateSufficientHightLocation calculates location correctly 4`] = `48`;

exports[`spinnerSegmentHelper calculateSufficientHightLocation calculates location correctly 5`] = `48`;

exports[`spinnerSegmentHelper calculateSufficientHightLocation calculates location correctly 6`] = `56.80023279893367`;

exports[`spinnerSegmentHelper calculateSufficientHightLocation calculates location correctly 7`] = `92.88393099705036`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 1`] = `"rotate(360deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 2`] = `"rotate(180deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 3`] = `"rotate(360deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 4`] = `"rotate(120deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 5`] = `"rotate(240deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 6`] = `"rotate(360deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 7`] = `"rotate(90deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 8`] = `"rotate(180deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 9`] = `"rotate(270deg)"`;

exports[`spinnerSegmentHelper createSegmentRotationString creates correct rotation strings 10`] = `"rotate(360deg)"`;
